guides/tutorials/codelabs/pubsub-to-appengine/index.md

```diff
@@ -97,13 +97,41 @@ Now configure Spinnaker to receive messages from your Google Cloud Pub/Sub subsc
     Spinnaker understands. The JSON snippet above defines the mapping specific to the GCS Pub/Sub message, but these are entirely
     user-supplied and can specify any valid Jinja transformation.
 
+    Here is an example message payload from uploading an object to a GCS bucket:
+
+    ```
+    {
+      "kind": "storage#object",
+      "id": "gcs-pub-sub/app.tar/1511803705417483",
+      "selfLink": "https://www.googleapis.com/storage/v1/b/gcs-pub-sub/o/app.tar",
+      "name": "app.tar",
+      "bucket": "gcs-pub-sub",
+      "generation": "1511803705417483",
+      "metageneration": "1",
+      "contentType": "application/x-tar",
+      "timeCreated": "2017-11-27T17:28:25.218Z",
+      "updated": "2017-11-27T17:28:25.218Z",
+      "storageClass": "MULTI_REGIONAL",
+      "timeStorageClassUpdated": "2017-11-27T17:28:25.218Z",
+      "size": "20480",
+      "md5Hash": "K8fipg9xurPwrBEEfrkP9w==",
+      "mediaLink": "https://www.googleapis.com/download/storage/v1/b/gcs-pub-sub/o/app.tar?generation=1511803705417483&alt=media",
+      "crc32c": "T4DRpw==",
+      "etag": "CIu+09aj39cCEAE="
+    }
+    ```
+
+    Any of the keys present in this message can be used in the transformation definition. Jinja is expressive, so you can even include
+    things like loops, array indices, and paths to sub-objects in your template. The only constraint on the template file is that
+    the resultant object must match [the artifact model class](https://github.com/spinnaker/kork/blob/master/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/model/Artifact.java),
+    and that one must be supplied to Spinnaker.
+
     Store the path to 'gcs-jinja.json' as an environment variable for the next step:
```

Comment:
With the temlate and the sample payload json, plus a few sentences after, there's a lot of stuff between step 2 and step 3.  Reader might get a little lost. One small improvement might be to make the "Store the path to..." and "Make sure the file permissions..." are substeps, a. and b. That will at least highlight them as things to do, and descrease the risk they'll get missed as just a couple more paragraphs of explanation.

Reply:
Clarified that section with substeps.